game/src/core/game/WindowEventManager.java

```diff
@@ -0,0 +1,292 @@
+  }
+
+  /**
+   * Removes a files dropped event listener.
+   *
+   * @param listener The listener to remove
+   * @return true if the listener was found and removed, false otherwise
+   */
+  public static boolean unregisterFilesDroppedListener(WindowFilesDroppedListener listener) {
+    return filesDroppedListeners.remove(listener);
+  }
+
+  /**
+   * Removes a window refresh request event listener.
+   *
+   * @param listener The listener to remove
+   * @return true if the listener was found and removed, false otherwise
+   */
+  public static boolean unregisterWindowRefreshListener(Runnable listener) {
+    return windowRefreshListeners.remove(listener);
+  }
+
+  /**
+   * Clears all registered window event listeners. This should be called when shutting down the
+   * application or when window event handling is no longer needed.
+   */
+  public static void clearAllListeners() {
+    windowCreatedListeners.clear();
+    windowIconifiedListeners.clear();
+    windowMaximizedListeners.clear();
+    windowFocusListeners.clear();
+    closeRequestedListeners.clear();
+    filesDroppedListeners.clear();
+    windowRefreshListeners.clear();
+  }
+
+  /**
+   * Gets the window listener implementation that handles all registered event listeners. This
+   * method is intended to be used by the window creation system.
+   *
+   * @return The window listener implementation
+   */
+  public static Lwjgl3WindowListener windowListener() {
+    return windowListener;
+  }
+
+  /**
+   * Creates a window listener implementation that dispatches events to all registered listeners.
+   *
+   * @return A window listener implementation
+   */
+  private static Lwjgl3WindowListener createWindowListener() {
+    return new Lwjgl3WindowListener() {
+      @Override
+      public void created(Lwjgl3Window window) {
+        for (WindowCreatedListener listener : windowCreatedListeners) {
+          listener.onWindowCreated(window);
+        }
+      }
+
+      @Override
+      public void iconified(boolean isIconified) {
+        for (Consumer<Boolean> listener : windowIconifiedListeners) {
+          listener.accept(isIconified);
+        }
+      }
+
+      @Override
+      public void maximized(boolean isMaximized) {
+        for (Consumer<Boolean> listener : windowMaximizedListeners) {
+          listener.accept(isMaximized);
+        }
+      }
+
+      @Override
+      public void focusLost() {
+        for (Consumer<Boolean> listener : windowFocusListeners) {
+          listener.accept(false);
+        }
+      }
+
+      @Override
+      public void focusGained() {
+        for (Consumer<Boolean> listener : windowFocusListeners) {
+          listener.accept(true);
+        }
+      }
+
+      @Override
+      public boolean closeRequested() {
+        // If no listeners are registered, allow the window to close by default
+        if (closeRequestedListeners.isEmpty()) {
+          return true;
+        }
+
```

Comment:
```suggestion
```

Reply:
Dieser Code ist redundant, der Fall ist bereits in der `for`-Loop bzw. dem Stream-Vorschlag eingeschlossen.